network/p2p/inspector/validation/control_message_validation_inspector_test.go

```diff
@@ -0,0 +1,450 @@
+		// set high cache miss threshold to ensure we only disseminate notification when it is exceeded
+		inspector.config.IWantRPCInspectionConfig.CacheMissThreshold = .9
+		msgIds := unittest.IdentifierListFixture(100).Strings()
+		expectedPeerID := peer.ID("peerID987654321")
+		inspectMsgReq, err := NewInspectRPCRequest(expectedPeerID, unittest.P2PRPCFixture(unittest.WithIWants(unittest.P2PRPCIWantFixture(msgIds))))
+		require.NoError(t, err, "failed to get inspect message request")
+
+		checkNotif := checkNotifFunc(t, expectedPeerID, p2pmsg.CtrlMsgIWant, IsIWantCacheMissThresholdErr)
+		distributor.On("Distribute", mock.AnythingOfType("*p2p.InvCtrlMsgNotif")).Return(nil).Once().Run(checkNotif)
+		rpcTracker.On("LastHighestIHaveRPCSize").Return(int64(100)).Maybe()
+		// return false each time to eventually force a notification to be disseminated when the cache miss count finally exceeds the 90% threshold
+		rpcTracker.On("WasIHaveRPCSent", mock.AnythingOfType("string")).Return(false).Run(func(args mock.Arguments) {
+			id, ok := args[0].(string)
+			require.True(t, ok)
+			require.Contains(t, msgIds, id)
+		})
+		require.NoError(t, inspector.processInspectRPCReq(inspectMsgReq))
+	})
+}
+
+// TestNewControlMsgValidationInspector_validateClusterPrefixedTopic ensures cluster prefixed topics are validated as expected.
+func TestNewControlMsgValidationInspector_validateClusterPrefixedTopic(t *testing.T) {
+	t.Run("validateClusterPrefixedTopic should not return an error for valid cluster prefixed topics", func(t *testing.T) {
+		inspector, distributor, _, idProvider, sporkID := inspectorFixture(t)
+		defer distributor.AssertNotCalled(t, "Distribute")
+		clusterID := flow.ChainID(unittest.IdentifierFixture().String())
+		clusterPrefixedTopic := channels.Topic(fmt.Sprintf("%s/%s", channels.SyncCluster(clusterID), sporkID))
+		from := peer.ID("peerID987654321")
+		idProvider.On("ByPeerID", from).Return(unittest.IdentityFixture(), true).Once()
+		require.NoError(t, inspector.validateClusterPrefixedTopic(from, clusterPrefixedTopic, flow.ChainIDList{clusterID, flow.ChainID(unittest.IdentifierFixture().String()), flow.ChainID(unittest.IdentifierFixture().String())}))
+	})
+
+	t.Run("validateClusterPrefixedTopic should not return error if cluster prefixed hard threshold not exceeded for unknown cluster ids", func(t *testing.T) {
+		inspector, distributor, _, idProvider, sporkID := inspectorFixture(t)
+		defer distributor.AssertNotCalled(t, "Distribute")
+		clusterID := flow.ChainID(unittest.IdentifierFixture().String())
+		clusterPrefixedTopic := channels.Topic(fmt.Sprintf("%s/%s", channels.SyncCluster(clusterID), sporkID))
+		from := peer.ID("peerID987654321")
+		idProvider.On("ByPeerID", from).Return(unittest.IdentityFixture(), true).Once()
+		require.NoError(t, inspector.validateClusterPrefixedTopic(from, clusterPrefixedTopic, flow.ChainIDList{flow.ChainID(unittest.IdentifierFixture().String())}))
```

Comment:
Please improve the test to check for `hard threshold - 1` many unknown cluster ids would not cause a notification.